traits/has_traits.py

```diff
@@ -115,6 +115,28 @@ class AbstractViewElement(abc.ABC):
 # Generic 'Any' trait:
 any_trait = Any().as_ctrait()
 
+# Prefixes that are not allowed for user defined traits.
+DisallowedNamePrefixes = (
+    "trait",
+    "_trait"
+)
+
+
+# A custom warning issued by Traits
+class TraitsWarning(Warning):
+    """ A warning from Traits which could indicate that something
+    in the user code is using traits in a way that is was not
+    designed to be used.
+    """
+    pass
+
+
+def _is_disallowed_prefix(name):
+    for prefix in DisallowedNamePrefixes:
```

Comment:
No need for a loop here: you can do `return name.startswith(DisallowedNamePrefixes)`.